### PR TITLE
Bump chart API version to v2

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Bump chart API version to v2 and migrate dependencies from `requirements.yaml` to `Chart.yaml`

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Review by sumo-drosiek — COMMENTED** [review]
Lets add https://helm.sh/docs/topics/charts/#chart-types

File: deploy/helm/sumologic/Chart.yaml [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+    version: 1.5.4
+    repository: https://falcosecurity.github.io/charts
+    condition: falco.enabled
+  - name: metrics-server
+    version: 5.0.2
+    repository: https://charts.bitnami.com/bitnami
+    condition: metrics-server.enabled
+  - name: telegraf-operator
+    version: 1.1.6
+    repository: https://sumologic.github.io/influxdata-helm-charts
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/Chart.yaml:33]: please update due to #1147

Comment (sumo-drosiek) [inline deploy/helm/sumologic/Chart.yaml:33]:
```suggestion
    version: 1.1.5
    repository: https://helm.influxdata.com
```

Comment (pmalek-sumo) [reply deploy/helm/sumologic/Chart.yaml]: done

**Review by sumo-drosiek — APPROVED** [review]
🚀

**Pull request merged**